apps/docs/docs/dev/09-expressions-and-operators.md

```diff
@@ -0,0 +1,148 @@
+}
+
+%% Uses spaces in the name so it is unique
+class ` :BinaryOperator` {
+    operator = '+'
+}
+
+class `:UnaryOperator` {
+    operator = '-'
+}
+
+%% Uses spaces in the name so it is unique
+class ` :NumericLiteral` {
+    value = 3
+}
+
+class `  :NumericLiteral` {
+    value = 5
+}
+
+class `:NumericLiteral` {
+    value = 7
+}
+
+`:BinaryOperator` --> ` :BinaryOperator` : leftOperand
+`:BinaryOperator` --> `:NumericLiteral` : rightOperand
+` :BinaryOperator` --> `:UnaryOperator` : leftOperand
+`:UnaryOperator` --> ` :NumericLiteral` : operand
+` :BinaryOperator` --> `  :NumericLiteral` : rightOperand
+```
+
+The AST is constructed in a way that ensures an unambiguous evaluation order when using depth-first search.
+This implies that the AST structure already reflects the precedence and associativity of the operators, and that parentheses do not need to be explicitly represented.
+
+For more details on how such a grammar for expressions can be realized, see the [official Langium documentation](https://langium.org/docs/grammar-language/#tree-rewriting-actions), [this blog post](https://www.typefox.io/blog/parsing-expressions-with-xtext) and the following two sections which discuss the precedence and associativity of operators.
+
+### Precedence of operators
+
+The grammar implicitly defines the precedence of operators.
+The operator precedence defines the order in which operators are evaluated within an expression.
+For example, multiplication has a higher precedence than addition, which leads to multiplications being performed before additions.
+In order to manually override such precedence conventions, parentheses can be used in expressions.
+
+The diagram below shows a more extensive precedence hierarchy, focused on common arithmetic operators.
+Note that the hierarchy is arranged in ascending order, according to the operator precedence.
+Such an order is similar to how operators in the actual Jayvee grammar are arranged:
+
+```mermaid
+graph TD
+  subgraph BinaryOperators
+    add/sub(Addition Operator / Subtraction operator) --> mul/div(Multiplication Operator / Division Operator)
+    mul/div --> exp/root(Exponential Operator / Root Operator)
+  end
+  subgraph UnaryOperators
+    exp/root --> plus/minus(Plus Operator / Minus Operator)
+  end
+  plus/minus --> literal(Numeric Literal)
+```
+
+In the Jayvee grammar, every level of precedence is represented by a single grammar rule.
+Within each such rule, the grammar rule which defines the operators with the next higher precedence is referenced.
+E.g. the `AdditiveExpression` rule refers to the `MultiplicativeExpression` rule because multiplicative operators are supposed to have the next higher precedence.
+
+In order to alter the precedence of operators, their hierarchy of grammar rules has to be adjusted accordingly in [`expression.langium`](https://github.com/jvalue/jayvee/blob/main/libs/language-server/src/grammar/expression.langium).
+
+### Associativity of operators
+
```

Comment:
I would have expected a statement of which way we designed this in Jayvee.

Reply:
We currently only have left-associative operators, but I think this will most likely change in the future (e.g. for the `pow` operator). To avoid the docs getting out of sync soon, I decided to not mention the current operator associativities here.

What do you think about pointing readers to `expression.langium` to look for the concrete associativities of operators, instead of listing them directly in the docs?